finance-service-api/src/main/kotlin/com/ajaxproject/financeservice/dto/FinanceResponse.kt

```diff
@@ -0,0 +1,66 @@
+package com.ajaxproject.financeservice.dto
+
+import com.ajaxproject.financeservice.enums.Finance
+import com.ajaxproject.financeservice.models.MongoFinance
+import com.ajaxproject.internalapi.finance.commonmodels.FinanceMessage
+import com.ajaxproject.internalapi.finance.commonmodels.FinanceType
+import java.util.*
+
+data class FinanceResponse(
+    val amount: Double,
+    val description: String,
+    val financeType: String,
+    val date: Date,
+) {
+
+    override fun toString(): String {
+        return """
+            |$financeType details:
+            |Amount: $amount
+            |Description: $description
+            |Date: $date
+        """.trimMargin()
+    }
+}
+
+fun MongoFinance.toFinanceResponse() = FinanceResponse(
+    amount = amount,
+    description = description,
+    financeType = financeType.toString(),
+    date = date
+)
+
+fun MongoFinance.toProtoFinance(): FinanceMessage {
+    return FinanceMessage.newBuilder()
+        .setUserId(userId)
+        .setFinanceType(financeType.toProtoEnumFinance())
+        .setAmount(amount)
+        .setDescription(description)
+        .build()
+}
+
+fun FinanceMessage.toMongoFinance(): MongoFinance {
+    return MongoFinance(
+        userId = userId,
+        financeType = financeType.toFinanceEnum(),
+        amount = amount,
+        description = description,
+        date = Date(),
+    )
+}
+
+
+fun Finance.toProtoEnumFinance(): FinanceType {
+    return when (this) {
+        Finance.INCOME -> FinanceType.INCOME
+        Finance.EXPENSE -> FinanceType.EXPENSE
+    }
+}
```

Comment:
fun FinanceType.toFinanceEnum(): Finance=
when (this) {
FinanceType.INCOME -> Finance.INCOME
FinanceType.EXPENSE -> Finance.EXPENSE
else -> throw IllegalArgumentException("Unknown finance type")
}